### PR TITLE
New and updated API product docs

### PR DESCRIPTION
**Pull request by wildmanonline**
This PR includes all of the new and updated guides for the Linode API. All guides that were previously under the Guides > Platform section and now with the Product section. This PR also includes some changes to the Linode CLI product docs.

### CODE REVIEW

File: docs/products/tools/api/_index.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+{{< note >}}
+To view the official documentation on each Linode API endpoint, see the [API developer documentation](/docs/api/).
+{{< /note >}}
+
+## Features
+
+### Build the Apps You Want
+
+Create your own dev tools with the Linode API. Build scripts and applications to automatically handle repeatable tasks. The Linode API provides access to every part of the Linode platform. Deploy Kubernetes clusters, attach block or object storage volumes, configure NodeBalancers, manage users, and more. All response data is returned in JSON for easy parsing and filtering.
```
Comment (andystevensname) [inline docs/products/tools/api/_index.md:23]: Object storage doesn't have volumes and they're not attached.

File: docs/products/tools/api/get-started/index.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+```
+
+## Get Configuration Parameters
+
+Specify the type, region, and image for the new Linode.
+
+1. Review the list of available images:
+
+    ```command
+    curl https://api.linode.com/v4/images/ | json_pp
```
Comment (andystevensname) [inline docs/products/tools/api/get-started/index.md:62]: We use json_pp a lot in these instructions, do we have any language stating that you'll have to download and install this package?

File: docs/products/tools/api/guides/block-storage/index.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+1. List the Compute Instances on your account:
+
+    ```command
+    curl -H "Authorization: Bearer $token" \
+        https://api.linode.com/v4/linode/instances
+    ```
+
+    Choose a Compute Instance from the returned list and copy its `id` and `region` values.
+
+1. Create a Volume in the same region as the target instance. Use the ID of the target instance and adjust the size, region, and label to the desired values:
```
Comment (andystevensname) [inline docs/products/tools/api/guides/block-storage/index.md:41]: Should we mention that size is in GB?

**Review by andystevensname — APPROVED** [review]

File: docs/products/tools/api/guides/block-storage/index.md (hunk 3 above)
Comment (wildmanonline) [reply docs/products/tools/api/guides/block-storage/index.md:41]: Done!

**Pull request merged**